controllers/configurationpolicy_controller.go

```diff
@@ -3180,6 +3170,15 @@ func (r *ConfigurationPolicyReconciler) updatePolicyStatus(
 
 			log.Info(fmt.Sprintf("Failed to update policy status. Retrying (attempt %d/%d): %s", i, maxRetries, err))
 		} else {
+			if sendEvent {
```

Comment:
So I think that if we only send the compliance event after the status has been updated, then if the event fails to be sent, it will *not* be sent on the next reconcile, because the status "looks correct".

The issue in this case (I believe) was that an event was sent, but then the status failed to be updated, and then when the policy re-reconciled, the cluster state matched the *old* status, so it didn't send another new compliance event like it maybe should've. (If the status update had worked, then the next reconcile would have been trying to send a Compliant compliance event, and update the status again).

Does that make sense?

One solution we've been considering is just to have the framework directly watch the policy status, as opposed to the compliance events. The problem is kind of caused by there being two sources of truth: the events, *and* the policy status. If we just switched to having one, we might prevent problems like this.

Reply:
policy_status_sync controller watch config, operator, cert status changes? Is my understanding correct?

Reply:
Also "So I think that if we only send the compliance event after the status has been updated, then if the event fails to be sent, it will not be sent on the next reconcile, because the status "looks correct" We can send events always when status is updated.